src/main/java/org/prebid/server/floors/BasicPriceFloorEnforcer.java

```diff
@@ -0,0 +1,213 @@
+        return auctionParticipation.with(bidderResponse.with(bidderSeatBid));
+    }
+
+    private static boolean isEnforcedDealFloors(ExtRequestPrebidFloors floors,
+                                                AccountPriceFloorsConfig accountPriceFloorsConfig) {
+
+        final ExtRequestPrebidFloorsEnforcement enforcement = ObjectUtil.getIfNotNull(floors,
+                ExtRequestPrebidFloors::getEnforcement);
+        final Boolean requestEnforceDealFloors = ObjectUtil.getIfNotNull(enforcement,
+                ExtRequestPrebidFloorsEnforcement::getFloorDeals);
+
+        final Boolean accountEnforceDealFloors = accountPriceFloorsConfig.getEnforceDealFloors();
+
+        return BooleanUtils.isTrue(requestEnforceDealFloors) && BooleanUtils.isTrue(accountEnforceDealFloors);
+    }
+
+    private BigDecimal resolveFloor(BidderBid bidderBid,
+                                    BidRequest bidderBidRequest,
+                                    BidRequest bidRequest) {
+
+        final PriceFloorInfo priceFloorInfo = bidderBid.getPriceFloorInfo();
+        final BigDecimal customBidderFloor = ObjectUtil.getIfNotNull(priceFloorInfo, PriceFloorInfo::getFloor);
+        if (customBidderFloor != null) {
+            return convertIfRequired(customBidderFloor, priceFloorInfo.getCurrency(), bidderBidRequest, bidRequest);
+        }
+
+        final Imp imp = correspondingImp(bidderBid.getBid(), bidderBidRequest.getImp());
+        return convertIfRequired(imp.getBidfloor(), imp.getBidfloorcur(), bidderBidRequest, bidRequest);
+    }
+
+    /**
+     * Converts floor according to incoming auction request currency (since response bid is already converted).
+     * <p>
+     * Floor currency resolved in order: floor currency -> bidder request currency -> auction request currency.
+     */
+    private BigDecimal convertIfRequired(BigDecimal floor,
+                                         String floorCurrency,
+                                         BidRequest bidderBidRequest,
+                                         BidRequest bidRequest) {
+
+        final String resolvedFloorCurrency = ObjectUtils.defaultIfNull(floorCurrency,
+                resolveBidRequestCurrency(bidderBidRequest));
+
+        final String bidRequestCurrency = resolveBidRequestCurrency(bidRequest);
+
+        if (resolvedFloorCurrency != null && !resolvedFloorCurrency.equals(bidRequestCurrency)) {
+            return currencyConversionService.convertCurrency(
+                    floor,
+                    bidRequest,
+                    resolvedFloorCurrency,
+                    bidRequestCurrency);
+        }
+
+        return floor;
+    }
+
+    private static String resolveBidRequestCurrency(BidRequest bidRequest) {
+        final List<String> currencies = bidRequest.getCur();
+        return CollectionUtils.isEmpty(currencies) ? null : currencies.get(0);
```

Comment:
Nitpick. \n before return. Also pls check similar

Reply:
But return it is more like separate part which shows what will be given outside the method